src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -295,6 +307,68 @@ public class HubEndPoint<THub, TClient> : EndPoint, IInvocationBinder where THub
             }
         }
 
+        private bool IsChannel(Type type, out Type payloadType)
+        {
+            var channelType = type.AllBaseTypes().FirstOrDefault(t => t.IsGenericType && t.GetGenericTypeDefinition() == typeof(ReadableChannel<>));
```

Comment:
To be fair, the type doesn't have to wait for C# 8, and wouldn't require a new netstandard version.